Dockerfile

```diff
@@ -11,18 +11,16 @@ WORKDIR /opt/atomicapp
 
 # add all of Atomic App's files to the container image
 ADD atomicapp/ /opt/atomicapp/atomicapp/
-ADD setup.py /opt/atomicapp/
-ADD requirements.txt /opt/atomicapp/
+ADD setup.py VERSION NULECULE requirements.txt /opt/atomicapp/
 
 # add EPEL repo for pip
 RUN echo -e "[epel]\nname=epel\nenabled=1\nbaseurl=https://dl.fedoraproject.org/pub/epel/7/x86_64/\ngpgcheck=0" > /etc/yum.repos.d/epel.repo
 
 # lets install pip, and gcc for the native extensions
 # and remove all after use
-RUN yum install -y --setopt=tsflags=nodocs python-pip python-setuptools docker gcc && \
-    python setup.py install && \
-    pip install -r ./requirements.txt && \
-    yum remove -y gcc cpp glibc-devel glibc-headers kernel-headers libmpc mpfr python-pip && \
+RUN yum install -y --setopt=tsflags=nodocs python-pip python-setuptools docker gcc
+RUN python setup.py install
+RUN yum remove -y gcc cpp glibc-devel glibc-headers kernel-headers libmpc mpfr python-pip && \
```

Comment:
caching seems to be irrelevant for a base image, I would go for small size rather than the time needed to build to image.